### PR TITLE
[android-ndk/r25] add 2022 LTS version

### PR DESCRIPTION
**Pull request by Bigpet**
Specify library name and version:  **android-ndk/r25**

This is an LTS version that works on M1 Macs, this version will be the only one until the 2023 release. 
So this version seems relevant to me.

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Bigpet** [comment]
looks like the permission fixing stuff got broken, the `clang++` 'binary' is literally just 5Byte big text file containing the 'clang'

**Review by wolfee001 — APPROVED** [review]
changes: ok  
if you feel the energy, you could fix the issues found by the v2 linter

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SSE4 — APPROVED** [review]

File: recipes/android-ndk/all/conanfile.py [hunk 1]
```diff
@@ -132,22 +132,24 @@ def _fix_permissions(self):
...
-                         sig == [0xFE, 0xEF, 0xFA, 0xCE] or \
-                         sig == [0xCE, 0xFA, 0xED, 0xFE]:
+                    elif sig in (
+                        [0xCA, 0xFE, 0xBA, 0xBE],
+                        [0xBE, 0xBA, 0xFE, 0xCA],
+                        [0xFE, 0xED, 0xFA, 0xCF],
+                        [0xCF, 0xFA, 0xED, 0xFE],
+                        [0xFE, 0xEF, 0xFA, 0xCE],
+                        [0xCE, 0xFA, 0xED, 0xFE]
+                    ):
```
Comment (SSE4) [inline recipes/android-ndk/all/conanfile.py:152]: much better, here and above 👍

**Pull request merged**